staging/src/k8s.io/client-go/util/jsonpath/jsonpath.go

```diff
@@ -276,6 +276,27 @@ func (j *JSONPath) evalIdentifier(input []reflect.Value, node *IdentifierNode) (
 	return results, nil
 }
 
+func (j *JSONPath) evalMap(input []reflect.Value, hasAsterisk bool) ([]reflect.Value, error) {
+	result := []reflect.Value{}
+	if !hasAsterisk {
+		return nil, fmt.Errorf("invalid jsonpath for objects. Index should be an asterisk when applied over an object")
+	}
+
+	for _, value := range input {
+		value, isNil := template.Indirect(value)
+		if isNil {
+			continue
+		}
+		if value.Kind() == reflect.Map {
+			for _, e := range value.MapKeys() {
```

Comment:
are these returned in deterministic order?

Reply:
no, they're not. https://golang.org/pkg/reflect/#Value.MapKeys:

> MapKeys returns a slice containing all the keys present in the map, in unspecified order.

this needs tests ensuring the output is deterministic (which should fail on this non-deterministic version)

Reply:
Done. The recently pushed code makes the keys ordered and then values are captured.